core/src/main/java/hudson/Functions.java

```diff
@@ -1082,29 +1078,55 @@ public static String getFooterURL() {
      * Like {@link #getSortedDescriptorsForGlobalConfig(Predicate)} but with a constant truth predicate, to include all descriptors.
      */
     public static Collection<Descriptor> getSortedDescriptorsForGlobalConfig() {
-        return getSortedDescriptorsForGlobalConfig(Predicates.<GlobalConfigurationCategory>alwaysTrue());
+        return getSortedDescriptorsForGlobalConfig(descriptor -> true);
     }
 
     /**
      * @deprecated This is rather meaningless.
      */
     @Deprecated
     public static Collection<Descriptor> getSortedDescriptorsForGlobalConfigNoSecurity() {
-        return getSortedDescriptorsForGlobalConfig(Predicates.not(GlobalSecurityConfiguration.FILTER));
+        return getSortedDescriptorsForGlobalConfigUnclassified();
     }
 
     /**
-     * Like {@link #getSortedDescriptorsForGlobalConfig(Predicate)} but for unclassified descriptors only.
+     * Descriptors in the global configuration form that users with {@link Jenkins#MANAGE} permission can configure.
+     *
      * @since 1.506
      */
     public static Collection<Descriptor> getSortedDescriptorsForGlobalConfigUnclassified() {
-        return getSortedDescriptorsForGlobalConfig(new Predicate<GlobalConfigurationCategory>() {
-            public boolean apply(GlobalConfigurationCategory cat) {
-                return cat instanceof GlobalConfigurationCategory.Unclassified;
+        return getSortedDescriptorsForGlobalConfig(d -> d.getCategory() instanceof GlobalConfigurationCategory.Unclassified && Jenkins.get().hasPermission(d.getRequiredGlobalConfigPagePermission()));
+    }
+
+    /**
+     * Descriptors shown in the global configuration form to users with {@link Jenkins#SYSTEM_READ} permission.
+     *
+     * @return
+     * @since TODO
+     */
+    @Restricted(NoExternalUse.class)
+    public static Collection<Descriptor> getSortedDescriptorsForGlobalConfigUnclassifiedReadable() {
+        return getSortedDescriptorsForGlobalConfig(d -> d.getCategory() instanceof GlobalConfigurationCategory.Unclassified && (
+                Jenkins.get().hasPermission(d.getRequiredGlobalConfigPagePermission()) || Jenkins.get().hasPermission(Jenkins.SYSTEM_READ)));
+    }
+
+    @Restricted(NoExternalUse.class)
+    public static void checkAnyPermission(AccessControlled ac, Permission[] permissions) {
+        if (permissions == null) {
+            return;
+        }
+
+        boolean failed = true;
+        for (Permission permission : permissions) {
+            if (ac.hasPermission(permission)) {
+                failed = false;
             }
-        });
+        }
+        if (failed) {
+            ac.checkPermission(permissions[0]);
```

Comment:
`ArrayIndexOutOfBOundsException` if passed in a zero length array.

Reply:
https://github.com/jenkinsci/jenkins/pull/4506/files#diff-841df66889e3c3214c68de134dfe05beR1115

changed it earlier